http/grafana.go

```diff
@@ -100,21 +99,38 @@ type timeSeriesOutput struct {
 	Target     string          `json:"target"`
 }
 
-func simpleJSONRequestFactory(method string, request []byte) (simpleJSONRequestInterface, error) {
-	var retval simpleJSONRequestInterface
+func simpleJSONRequestFactory(method string, request []byte) ([]simpleJSONRequestInterface, error) {
+	var reqBase requestSimpleJSONBase
+	if err := json.Unmarshal(request, &reqBase); err != nil {
+		return nil, err
+	}
+	var requests []simpleJSONRequestInterface
 	switch method {
 	case "query":
-		retval = &simpleJSONQueryRequest{Backend: defaultBackend}
+		for _, target := range reqBase.Targets {
+			currRequest := &simpleJSONQueryRequest{Backend: defaultBackend, requestSimpleJSONBase: reqBase}
+			currRequest.Type = target["type"].(string)
+			fieldInput := target["target"].(string)
```

Comment:
Doing it like that means that the service will panic, crash and restart on a bad input.